AsyncDisplayKit/Private/ASInternalHelpers.h

```diff
@@ -33,6 +33,26 @@ BOOL ASRunningOnOS7();
 ASDISPLAYNODE_EXTERN_C_END
 
 /**
+ @summary Conditionally performs UIView geometry changes in the given block without animation and call completion block afterwards.
+ 
+ Used primarily to circumvent UITableView forcing insertion animations when explicitly told not to via
+ `UITableViewRowAnimationNone`. More info: https://github.com/facebook/AsyncDisplayKit/pull/445
+ 
+ @param withoutAnimation Set to `YES` to perform given block without animation
+ @param block Perform UIView geometry changes within the passed block
+ @param completion Call completion block if UIView geometry changes within the passed block did complete
+ */
+ASDISPLAYNODE_INLINE void ASPerformBlockWithoutAnimationCompletion(BOOL withoutAnimation, void (^block)(), void (^completion)()) {
```

Comment:
Should this function name include WITHOUT or WITH completion?

This seems like a bit risky of an approach, but something that could be extremely beneficial for reducing the massive overhead in collection views that we are seeing from animations. Is it necessary to include this in this change? It seems like it should not be required for the upgrade to the loading system to work correctly, but if it is, please share more details so that I can properly understand the interaction